plugin/clang.vim

```diff
@@ -875,9 +935,20 @@ func! s:ClangExecute(root, clang_options, line, col)
   let l:command .= ' 1>'.l:tmps[0].' 2>'.l:tmps[1]
   let l:res = [[], []]
   if has("nvim")
-    let l:argv = ['sh', '-c', l:command]
-    call s:PDebug("s:ClangExecute::job.argv", l:argv, 2)
-    call jobstart(l:argv, {'fstdout': l:tmps[0], 'fstderr': l:tmps[1], 'on_exit': function('ClangExecuteNeoJobHandler')})
+    let b:clang_memtmps = [[],[]]
+
+    let l:nvimsrc = getline(1, '$')
+    call add(l:nvimsrc, "\<cr>")
+
+    let l:nvimcmd = printf('%s -cc1 -x c++ -fsyntax-only -code-completion-macros -code-completion-at -:%d:%d %s',
+                      \ g:clang_exec, a:line, a:col, a:clang_options)
+    let l:argv = l:nvimcmd
+
+    " set env
+    let l:jobid = jobstart(['sh', '-c', l:argv], s:neojobcallbacks)
```

Comment:
I have tested on my machine, you can just put callbacks in local var, *please* put them in a local dict, `s:neojobcallbacks` is not good.
e.g.
```viml
jobstart(l:argv, {'fstdout': l:tmps[0], 'fstderr': l:tmps[1], 'on_exit': function('ClangExecuteNeoJobHandler')})
```